tools/clang/include/clang/SPIRV/SpirvFunction.h

```diff
@@ -42,13 +42,29 @@ class SpirvFunction {
   // TODO: The responsibility of assigning the result-id of a function shouldn't
   // be on the function itself.
   uint32_t getResultId() const { return functionId; }
+
   // TODO: There should be a pass for lowering QualType to SPIR-V type,
   // and this method should be able to return the result-id of the SPIR-V type.
   // Both the return type of the function as well as the SPIR-V "function type"
   // are needed. SPIR-V function type (obtained by OpFunctionType) includes both
   // the return type as well as argument types.
-  uint32_t getReturnTypeId() const { return 0; }
-  uint32_t getFunctionTypeId() const { return 0; }
+  uint32_t getReturnTypeId() const { return returnTypeId; }
+  void setReturnTypeId(uint32_t id) { returnTypeId = id; }
+
+  // Sets the lowered (SPIR-V) function type.
+  void setReturnType(SpirvType *type) { spirvReturnType = type; }
+  // Returns the lowered (SPIR-V) function type.
+  SpirvType *getReturnType() const { return spirvReturnType; }
```

Comment:
We should use `const SpirvType*` here. The type should be immutable,  unlike instructions. We will do transformations on instructions, but not on types. Having mutable pointers sometimes invites subtle bugs. So unless it is required by the work needed to be done, we should always prefer immutable ones.